mlrun/config.py

```diff
@@ -284,6 +285,29 @@ def dbpath(self, value):
             # when dbpath is set we want to connect to it which will sync configuration from it to the client
             mlrun.db.get_run_db(value)
 
+    @property
+    def iguazio_api_url(self):
+        if not self._iguazio_api_url:
+            # we want to be able to run with old versions of the service who runs the API (which doesn't configure this
+            # value) so we're doing best effort to try and resolve it from other configurations
+            if self.httpdb.builder.docker_registry and self.igz_version:
+                docker_registry_url = self.httpdb.builder.docker_registry
```

Comment:
move to some helper `_extract_iguazio_api_from_docker_registry_url`
and use `urllib.parse` to parse out the host only

Reply:
In the practical scenario this `docker_registry_url` will be something like `docker-registry.default-tenant.app.hedingber-301-1.iguazio-cd2.com:80`, this lib doesn't like urls without schema, so in order for it work I ended up with:
```
        # add schema otherwise parsing go wrong
        if "://" not in docker_registry_url:
            docker_registry_url = f"http://{docker_registry_url}"
        parsed_registry_url = urllib.parse.urlparse(docker_registry_url)
```
Not the best, but 🤷‍♂️